src/utils/IterableActionRunner.ts

```diff
@@ -0,0 +1,62 @@
+import IterableConfig from 'src/utils/IterableConfig';
+import {
+  IterableAction,
+  IterableActionContext,
+  IterableActionSource
+} from '../embedded/types';
+
+class IterableActionRunnerImpl {
+  static executeAction(
+    context: any,
+    action: IterableAction | null,
+    source: IterableActionSource
+  ): boolean {
+    if (action === null) {
+      return false;
+    }
+
+    const actionContext: IterableActionContext = { action, source };
+    if (action.type === 'openUrl') {
+      return IterableActionRunnerImpl.openUri(action.data, actionContext);
+    } else {
+      return IterableActionRunnerImpl.callCustomActionIfSpecified(
+        action,
+        actionContext
+      );
+    }
+  }
+
+  private static openUri(
+    uri: string,
+    actionContext: IterableActionContext
+  ): boolean {
+    if (IterableConfig.urlHandler) {
+      if (IterableConfig.urlHandler.handleIterableURL(uri, actionContext)) {
+        return true;
+      }
+    }
+
+    window.open(uri, '_blank');
+
+    return true;
+  }
+
+  private static callCustomActionIfSpecified(
+    action: IterableAction,
+    actionContext: IterableActionContext
+  ): boolean {
+    if (action.type && action.type !== '') {
+      if (IterableConfig.customActionHandler) {
+        return IterableConfig.customActionHandler.handleIterableCustomAction(
+          action,
+          actionContext
+        );
+      }
+    }
+    return false;
+  }
+}
+
+export class IterableActionRunner extends IterableActionRunnerImpl {
+  static config = IterableConfig;
+}
```

Comment:
@hani-iterable What is this config needed for?

Reply:
No need for that. I have removed.